examples_tests/61.BlitFilterTest/blit/blit.glsl

```diff
@@ -0,0 +1,162 @@
+		const vec4 loadedData = nbl_glsl_blit_getData(inputPixelCoord).data * premultWeights;
+		for (uint ch = 0u; ch < _NBL_GLSL_BLIT_OUT_CHANNEL_COUNT_; ++ch)
+			scratchShared[ch][wgLocalWindowIndex * windowPixelCount + windowLocalPixelIndex] = loadedData[ch];
+	}
+	barrier();
+
+	for (uint ch = 0u; ch < _NBL_GLSL_BLIT_OUT_CHANNEL_COUNT_; ++ch)
+	{
+		const uvec3 stride = uvec3(1u, params.windowDim.x, params.windowDim.x * params.windowDim.y);
+
+		for (uint axis = 0u; axis < _NBL_GLSL_BLIT_DIM_COUNT_; ++axis)
+		{
+			const uint stride = stride[axis];
+			const uint elementCount = (windowPixelCount * params.windowsPerWG) / stride;
+
+			const uint adderLength = params.windowDim[axis];
+			const uint paddedAdderLength = roundUpToPoT(adderLength);
+			const uint adderCount = elementCount / adderLength;
+			const uint addersPerStep = _NBL_GLSL_WORKGROUP_SIZE_ / paddedAdderLength;
+			const uint adderStepCount = (adderCount + addersPerStep - 1) / addersPerStep;
+
+			for (uint adderStep = 0u; adderStep < adderStepCount; ++adderStep)
+			{
+				const uint stepLocalAdderIndex = gl_LocalInvocationIndex / paddedAdderLength;
+				const uint wgLocalAdderIndex = adderStep * addersPerStep + stepLocalAdderIndex;
+				const uint adderLocalPixelIndex = gl_LocalInvocationIndex % paddedAdderLength;
+
+				for (uint s = paddedAdderLength / 2u; s > 0u; s >>= 1u)
+				{
+					if ((adderLocalPixelIndex < s) && (stepLocalAdderIndex < addersPerStep) && (wgLocalAdderIndex < adderCount))
+					{
+						float addend = 0.f;
+						if (adderLocalPixelIndex + s < adderLength)
+							addend = scratchShared[ch][(wgLocalAdderIndex * adderLength + adderLocalPixelIndex + s) * stride];
+
+						scratchShared[ch][(wgLocalAdderIndex * adderLength + adderLocalPixelIndex) * stride] += addend;
+					}
+					barrier();
+				}
+			}
+		}
+		barrier();
+	}
+
+	for (uint step = 0u; step < stepCount; ++step)
+	{
+		const bool firstInvocationOfWindow = (gl_LocalInvocationIndex % windowPixelCount) == 0u ? true : false;
+		if (!firstInvocationOfWindow)
+			break;
+
+		const uint stepLocalWindowIndex = gl_LocalInvocationIndex / windowPixelCount;
+		if (stepLocalWindowIndex >= windowsPerStep) // otherwise some invocations in this step might interfere with next step's windows.
+			break;
+
+		const uint wgLocalWindowIndex = stepLocalWindowIndex + step * windowsPerStep;
+		if (wgLocalWindowIndex >= params.windowsPerWG) // otherwise some invocations in this workgroup might interfere with next workgroup's windows.
+			break;
+
+		const uint globalWindowIndex = gl_WorkGroupID.x * params.windowsPerWG + wgLocalWindowIndex;
+		if (globalWindowIndex >= totalWindowCount)
+			break;
+
+		nbl_glsl_blit_pixel_t dataToStore;
+		for (uint ch = 0u; ch < _NBL_GLSL_BLIT_OUT_CHANNEL_COUNT_; ++ch)
+			dataToStore.data[ch] = scratchShared[ch][wgLocalWindowIndex * windowPixelCount];
+
+		const uint bucketIndex = packUnorm4x8(vec4(dataToStore.data.a, 0.f, 0.f, 0.f));
```

Comment:
Using a `packUnorm4x8` tightly and invisibly couples us with using `DefaultAlphaBinCount == 256`. I think, we should derive `bucketIndex` from `DefaultAlphaBinCount` then we can increase the alpha bins to raise image (coverage) quality if it is required.

Reply:
okay, but remember the highest the bin count can be is ~4096 (12 bit alpha) elements without failing due to smem limitations

Reply:
hmm I think you also have some implicit assumptions about histogram (bin count size) and the number of invocations in the workgroup

Reply:
thats the only reason why I think so, basically your code is fine as long as number of workgroups>=alphaBinCount

wondering what will happen when you make the bin count 4096 and workgroups stay 256.

Reply:
> wondering what will happen when you make the bin count 4096 and workgroups stay 256.

It won't work correctly, since we don't reuse invocations in the normalization shader.
It will run considering only the first 256 elements of the 4096 element histogram.

Reply:
Resolved.